source/content.tsx

```diff
@@ -8,28 +8,28 @@ import {storage} from './api';
 let willPreviewFiles: boolean;
 let hideRegExp: RegExp;
 
-function overflowsParent(el): boolean {
-	return el.getBoundingClientRect().right > el.parentNode.getBoundingClientRect().right;
+function overflowsParent(el: Element): boolean {
+	return el.getBoundingClientRect().right > el.parentElement!.getBoundingClientRect().right;
 }
 
 function updateUI(): void {
 	const hiddenFiles =
 		select.all('.files .js-navigation-item .content > span > *')
-			.filter(el => hideRegExp.test(el.textContent));
+			.filter(el => hideRegExp.test(el.textContent!));
 
 	if (hiddenFiles.length === 0) {
 		return;
 	}
 
 	const hidden = document.createDocumentFragment();
-	let previewList: HTMLElement;
+	let previewList: HTMLElement | false = false;
```

Comment:
I would prefer using just `undefined` instead of `false`. Better to keep the type singular when possible.

Reply:
I had to keep `| undefined = undefined` here because without it I get this message:

```
[tsl] ERROR in /Users/bfred/Web/projects-extensions/hide-files-on-github/source/content.tsx(41,27)
      TS2454: Variable 'previewList' is used before being assigned.

[tsl] ERROR in /Users/bfred/Web/projects-extensions/hide-files-on-github/source/content.tsx(57,15)
      TS2454: Variable 'previewList' is used before being assigned.
```